src/main/java/org/jenkinsci/test/acceptance/plugins/matrix_auth/MatrixAuthorizationStrategy.java

```diff
@@ -20,17 +21,32 @@ public MatrixAuthorizationStrategy(GlobalSecurityConfig context, String path) {
      * Adds a new user to this matrix.
      */
     public MatrixRow addUser(String name) {
-        runThenHandleAlert(
-                () -> this.table
-                        .resolve()
-                        .findElement(
-                                by.xpath(
-                                        "../div/span/span/button[text()='Add user\u2026'] | ../div/button[text()='Add user\u2026']"))
-                        .click(),
-                a -> {
-                    a.sendKeys(name);
-                    a.accept();
-                });
+        try {
+            // 3.2.3 and later
+            this.table
+                    .resolve()
+                    .findElement(
+                            by.xpath(
+                                    "../div/span/span/button[text()='Add user\u2026'] | ../div/button[text()='Add user\u2026']"))
+                    .click();
+            getPage().find(by.css("dialog input")).sendKeys(name);
+            getPage().find(by.css("dialog .jenkins-button--primary")).click();
+        } catch (UnhandledAlertException ex) {
+            // 3.2.2 and earlier
```

Comment:
is there any particular reason you've added this? The versions that are supported by ATH passed in https://github.com/jenkinsci/acceptance-test-harness/runs/31935029697

Reply:
To clarify, we should not add compatibility code with earlier versions of plugins when we change something? I'm find with that, but that means we tie ATH versions (revisions?) really closely to plugin versions.

Reply:
A recent example: https://github.com/jenkinsci/credentials-plugin/pull/551#issuecomment-2395592063

Latest weekly and LTS is what we support here.

Compatibility changes across versions are either for LTS vs weekly or for forward compatibility generally.

Reply:
Now that I've already added the code, would you prefer I undo that, or is it fine to keep if it exists? Also note the latest commit that adds to the general API, potentially unnecessarily (now).

Reply:
fine to keep since its added I think and the general purpose code may be useful somewhere.